MAINTAINING.md

```diff
@@ -142,49 +142,78 @@ Internal APIs:
 
 ## Managing Issues
 
-### User Impacting Changes
-
-__RULE__: Any commit that may affect the end user in some way MUST have either a
-changelog entry OR MUST have an issue marked with one of the following labels
-OR both.  We can have more than one of these labels on the same issue e.g.
-breaking, enhancement:
-
-* breaking
-* deprecating
-* enhancement
-* performance
-* usability
-* documentation
-* bug
-
-Note that if you are making a big feature change you may have a single issue
-for that feature and attach many commits with it. So you do not necessarily
-need to have an issue for each commit.
-
-### Maintenance Changes
-
-Commits that do not impact the end user in any way are not required to have a
-changelog entry or an issue.  Issues that do not have a corresponding commit
-may be left without a label but preferably should be marked with one of the
-following:
+We label the issues in different dimensions based on the lifecyle and different
+management aspects of the issue. The folowing sections discuss the labels used
+to manage the issues. If a new label is required, it should be discussed before
+creating it.
+
+### Disposition
+
+The level-1 triaging of the issue determines the current disposition
+of the issue. If no change is required the issue must have one of the
+following labels:
 
+* invalid
 * question
```

Comment:
question and discussion can be combined to "Need More Info"

Reply:
We may want to bucket issues differently depending on whether it is a clarifying question or if it is a proposal/discussion about a change. Let's keep them separate.